src/panels/MissionsPanel/TimeLine/TimeLine.tsx

```diff
@@ -0,0 +1,327 @@
+  const transitionDuration = 750;
+
+  const clipPathTop = margin.top - paddingGraph.top;
+  const clipPathBottom = height - margin.bottom + 2 * paddingGraph.bottom;
+
+  const timeRange = [
+    new Date(missionOverview.timerange.start),
+    new Date(missionOverview.timerange.end)
+  ];
+  const xScale = scaleLinear()
+    .range([margin.left, width - margin.right])
+    .domain([0, nestedLevels]);
+  const yScale = scaleUtc([height - margin.bottom, margin.top]).domain(timeRange);
+
+  // Calculate axes
+  const yAxis = axisLeft(yScale);
+
+  // TODO: we dont need the xAxis since we're not plotting anything on it anyways
+  //   const xAxis = axisTop(xScale)
+  //     .tickFormat(() => '')
+  //     .tickSize(0)
+  //     .ticks(nestedLevels);
+
+  // TODO right now we don't have a way to get the size of the panel we're in so we can't
+  // do any updates related to a panel resize and therefore use a fixed height.
+  // When height changes of window, rescale y axis
+  /*
+  useEffect(() => {
+    // Update the axis every time window rescales
+    yScale = scaleUtc()
+    .range([height - margin.bottom, margin.top])
+    .domain(timeRange);
+    yAxis = axisLeft(yScale);
+    select(yAxisRef.current).call(yAxis);
+}, [height]);
+*/
+  // TODO: these use Effects should fill their dependency arrays but if we do it with eg
+  // yAxis its recomputed every time, so we need to do some memo? or somethign here
+  // On mount, style axes
+  useEffect(() => {
+    select(yAxisRef.current).call(yAxis);
+    // TODO set font family? Previously was 'Segoe UI' font on ticks
+    select(yAxisRef.current).selectAll('.tick text').style('font-size', '1.3em');
+  }, []);
+
+  // Add zoom and update it every the the y scale changes (TODO: panel resize)
+  useEffect(() => {
+    zoomRef.current = zoom<SVGElement, unknown>()
+      .on('zoom', (event) => {
+        // TODO: event is any here which is super annoying, try to find a way to get this
+        // typed properly..
+        const newYScale = event.transform.rescaleY(yScale);
+        select(yAxisRef.current).call(yAxis.scale(newYScale));
+        setScale(event.transform.k);
+        setTranslation(event.transform.y);
+      })
+      .scaleExtent(scaleExtent)
+      .extent(translateExtent)
+      .translateExtent(translateExtent);
+    select(svgRef.current).call(zoomRef.current);
+  }, [yScale]);
+
+  function reset() {
+    if (!zoomRef.current) {
+      return;
+    }
+    select(svgRef.current)
+      .transition()
+      .duration(transitionDuration)
+      .call(
+        zoomRef.current.transform,
+        zoomIdentity,
+        zoomTransform(select(svgRef.current).node()).invert([width * 0.5, height * 0.5])
+      );
+  }
+
+  function zoomByButton(zoomValue: number) {
+    if (!zoomRef.current) {
+      return;
+    }
+    select(svgRef.current)
+      .transition()
+      .duration(transitionDuration)
+      .call(zoomRef.current.scaleBy, zoomValue);
+  }
+
+  function centerTime() {
```

Comment:
This function has a weird "zooming out" motion that I find very confusing. I know that that's how it worked in the old UI, but the transition happenes in a way so that its hard to see what is going on with the time (with zooming and translation happening simultaneously)

I think that expected behavior for me would be that it just "scrolls" (translated) to where the time indicator is, without zooming. Or first zooms out, then moves to show the indicator, then zooms in